public/app/features/dashboard/dashgrid/DashboardPanel.tsx

```diff
@@ -108,6 +112,16 @@ export class DashboardPanel extends PureComponent<Props, State> {
     const angularPanel = loader.load(this.element, scopeProps, template);
 
     this.setState({ angularPanel });
+
+    this.loadDataSource();
+  }
+
+  async loadDataSource() {
+    const datasourceInstance = await getDatasourceSrv().get(this.props.panel.datasource, this.props.panel.scopedVars);
+    // TODO check if mounted
+    if (datasourceInstance.name === this.props.panel.datasource) {
+      this.setState({ datasourceInstance });
+    }
```

Comment:
Loads the datasource here as we need it in both Panel and PanelConfig right now.